README.md

```diff
@@ -22,268 +22,320 @@
+        sudo rm -rf /usr/local/go
+        wget https://dl.google.com/go/go1.14.4.linux-amd64.tar.gz
+        sudo tar -C /usr/local -zxvf go1.14.4.linux-amd64.tar.gz
     ```
-    **In step 2, the folder name should remain free5gc. Please do not modify it or the compilation would fail.**
 
-3. Compile network function services in `free5gc` individually, e.g. AMF (redo this step for each NF), or
+    * If Go is not installed on your system:
+
     ```bash
-    cd ~/free5gc
-    go build -o bin/amf -x src/amf/amf.go
+        wget https://dl.google.com/go/go1.14.4.linux-amd64.tar.gz
+        sudo tar -C /usr/local -zxvf go1.14.4.linux-amd64.tar.gz
+        mkdir -p ~/go/{bin,pkg,src}
+        echo 'export GOPATH=$HOME/go' >> ~/.bashrc
+        echo 'export GOROOT=/usr/local/go' >> ~/.bashrc
+        echo 'export PATH=$PATH:$GOPATH/bin:$GOROOT/bin' >> ~/.bashrc
+        source ~/.bashrc
     ```
-    **To build all network functions in one command**
+
+    * Further information and installation instructions for `golang` are available at the [official golang site](https://golang.org/doc/install).
+
+3. Control-plane Supporting Pacakges
+
+```bash
+sudo apt -y update
+sudo apt -y install mongodb wget git
+sudo systemctl start mongodb
+```
+
+4. User-plane Supporting Packages
+
+```bash
+sudo apt -y update
+sudo apt -y install git gcc cmake autoconf libtool pkg-config libmnl-dev libyaml-dev
+go get -u github.com/sirupsen/logrus
+```
+
+5. Linux Host Network Settings
+
+```bash
+sudo sysctl -w net.ipv4.ip_forward=1
+sudo iptables -t nat -A POSTROUTING -o <dn_interface> -j MASQUERADE
+sudo systemctl stop ufw
+```
+
+### B. Install Control Plane Elements
+    
+1. Clone the free5GC repository
+    * To install the latest stable build (v3.0.3):
+
     ```bash
-    ./build.sh
+        cd ~
+        git clone --recursive -b v3.0.3 -j `nproc` https://github.com/free5gc/free5gc.git
+        cd free5gc
     ```
 
+    * Alternatively, if you wish to install the latest nightly build:
 
-### C. Install User Plane Function (UPF)
-    
-1. Please check Linux kernel version if it is `5.0.0-23-generic`
     ```bash
-    uname -r
+        cd ~/free5gc
+        git checkout master
+        git submodule sync
+        git submodule update --init --jobs `nproc`
+        git submodule foreach git checkout master
+        git submodule foreach git pull --jobs `nproc`
     ```
 
+2. Install all Go module dependencies
+
+```bash
+cd ~/free5gc
+go mod download
+```
+**NOTE: the root folder name for this repository must be `free5gc`.  If it is changed, compilation will fail.**
+
+3. Compile network function services in `free5gc`
+    * To do so individually (e.g., AMF only):
 
-    Get Linux kernel module 5G GTP-U
     ```bash
-    git clone -b v0.1.0 https://github.com/PrinzOwO/gtp5g.git
-    cd gtp5g
-    make
-    sudo make install
+        cd ~/free5gc
+        go build -o bin/amf -x src/amf/amf.go
```

Comment:
We have update build procedure to `make amf`, `make smf`, `make all`, ...

Reply:
The building process still remain as the old way. It need to be using make to build it now.